js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -128,6 +128,38 @@ export const ERROR_INVALID_ASSET_SELECTION =
   ` the same code location and share a partition space, or form a connected` +
   ` graph in which root assets share the same partitioning.`;
 
+function materializationDisabledReason(all: Asset[], materializable: Asset[]) {
+  if (!all.length) {
+    return 'Select one or more assets to materialize';
+  }
+  if (all.some((a) => !a.hasMaterializePermission)) {
+    return 'You do not have permission to materialize assets';
```

Comment:
But you might be able to materialize *some* of them, no?

Reply:
Yeah I think that is true... It's possible this should behave like a mixed materializable / observable selection and say `Materialize (2)` instead of `Materialize (4)` when you are missing permissions on two of the assets. I'll ask Josh next time we talk to him, I'm not sure how common this case is!